server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduleService.java

```diff
@@ -500,31 +500,39 @@ public void terminate(ScheduleEntity scheduleConfig) throws SchedulerException {
         scheduleRepository.updateStatusById(scheduleConfig.getId(), ScheduleStatus.TERMINATED);
     }
 
+    @Transactional(rollbackFor = Exception.class)
+    public void innerTerminate(Long scheduleId) throws SchedulerException {
+        ScheduleEntity schedule = nullSafeGetById(scheduleId);
+        JobKey jobKey = QuartzKeyGenerator.generateJobKey(schedule);
+        quartzJobService.deleteJob(jobKey);
+        scheduleRepository.updateStatusById(schedule.getId(), ScheduleStatus.TERMINATED);
+    }
+
     /**
      * The method detects whether the database required for scheduled task operation exists. It returns
      * true and terminates the scheduled task if the database does not exist. If the database exists, it
      * returns false.
      */
     public boolean vetoJobExecution(Long scheduleId) {
-        Optional<ScheduleEntity> scheduleEntityOptional = scheduleRepository.findById(scheduleId);
-        if (scheduleEntityOptional.isPresent()) {
-            ScheduleEntity entity = scheduleEntityOptional.get();
-            boolean isInvalid = isInvalidSchedule(entity);
-            if (isInvalid) {
-                try {
-                    log.info(
-                            "The project or database for scheduled task operation does not exist, and the schedule is being terminated, scheduleId={}",
-                            scheduleId);
-                    terminate(entity);
-                } catch (Exception e) {
-                    log.warn("Terminate schedule failed,scheduleId={}", scheduleId);
-                }
+        Schedule schedule = null;
+        boolean isValidSchedule;
+        try {
+            schedule = nullSafeGetModelById(scheduleId);
+            isValidSchedule = isValidSchedule(schedule);
+        } catch (NotFoundException e) {
```

Comment:
what about other exception?

Reply:
task will be executed if catch other exception